### PR TITLE
[METAL-2989] Enable metrics scraping

### PR DESCRIPTION
**Pull request by dpeckett**

### CODE REVIEW

File: pkg/utils/proxy/cloudproxy.go [hunk 1]
```diff
@@ -111,6 +111,10 @@ func (cp *CloudProxy) deploymentSpec() (v1apps.DeploymentSpec, error) {
 		},
 		Template: v1.PodTemplateSpec{
 			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
```
Comment (dabde) [inline pkg/utils/proxy/cloudproxy.go:114]: should we do this as a condition to the instance monitoring option?

Comment (dpeckett) [reply pkg/utils/proxy/cloudproxy.go:114]: good point

File: pkg/utils/proxy/cloudproxy.go [hunk 2]
```diff
@@ -104,14 +106,24 @@ func (cp *CloudProxy) deploymentSpec() (v1apps.DeploymentSpec, error) {
 
 	terminationGracePeriodSeconds := int64(120) // force kill pod after this time
 
+	var annotations map[string]string
+
+	if cp.MonitoringEnabled {
+		annotations = map[string]string{
+			"prometheus.io/scrape": "true",
```
Comment (dabde) [inline pkg/utils/proxy/cloudproxy.go:113]: (minor)

prometheus is default to scape path "/metrics". But it's possible to overwrite this with this annotation `prometheus.io/path`.

but think, if somebody need this, it's a simple addition.

**Review by dabde — APPROVED** [review]

File: pkg/utils/proxy/cloudproxy.go (hunk 2 above)
Comment (dpeckett) [reply pkg/utils/proxy/cloudproxy.go:113]: Agreed, in the long run we could add a loosely typed proxyAnnotations map. But YAGNI anytime soon.

**Pull request merged**